cEP-0017.md

```diff
@@ -0,0 +1,366 @@
+            sys.stdout.write( "Please respond with 'yes' or 'no'" )
+```
+
+### Splitting `DistributionRequirement`
+
+```
+from coala_utils.decorators import generate_eq, generate_repr
+from sarge import run, Capture
+import distro
+from functools import lru_cache
+
+@generate_eq("platform", "package", "version", "repo")
+@generate_repr()
+class PlatformRequirement:
+    """
+    This class helps keeping track of bear distribution requirements.
+    """
+
+    REQUIREMENTS = {}
+
+    def __init__(self, platform: str, package: str, version="", repo=""):
+        """
+        :param platform:  A string with the name of the distribution. Currently,
+                          the supported platforms are Debian, Ubuntu
+                          CentOS, Fedora, Arch Linux, openSUSE/SUSE, Gentoo Linux,
+                          Darwin
+        :param package: A string with the name of the package to be installed.
+        :param version: A version string. Leave empty to specify latest version.
+        :param repo:    The repository from which the package is to be
+                        installed.
+        """
+        self.platform = platform
+        self.package = package
+        self.version = version
+        self.repo = repo
+
+    def __str__(self):
+        """
+        Just return package name, followed by version if given.
+        """
+        if self.version:
+            return "{}{} on {}".format(self.package,
+                                       self.version,
+                                       self.platform)
+        else:
+            return "{} on {}".format(self.package,
+                                     self.platform)
+
+    def install_package(self):
+        """
+        Runs the install command for the package given in a sub-process.
+        """
+        run(" ".join(self.install_command()), stdout=Capture(),
+            stderr=Capture())
+
+    def install_command(self):
+        """
+        Returns a string with the installation command, to be used by
+        "install_package()".
+
+        :raises NotImplementedError: Method is not implemented
+        """
+        raise NotImplementedError
+
+    def is_installed(self):
+        """
+        Check if the requirement is satisfied.
+
+        :return: Returns True if satisfied, False if not.
+        :raises NotImplementedError: Method is not implemented
+        """
+        raise NotImplementedError
+
+    @lru_cache()
+    def get_platform():
+        return platform.system(), distro.id()
+
+    def check_platform():
+        """
+        Check if the user is on the target platform
+
+        :return: Returns True if user is on correct platform for this Requirement,
+        otherwise false.
+        """
+        if get_platform()[0] is 'Darwin':
+            return self.platform == 'Darwin'
+        elif get_platform()[0] is 'Linux':
+            return get_platform()[1] == self.platform
+```
+
+## Ansible
+
+### Package Requirement
+
+```
```

Comment:
Syntax can be highlighted using ```yaml

e.g
```yaml
- pip:
    name: package_name
    version: 0.11
```